metrics-core/src/main/java/com/codahale/metrics/ScheduledReporter.java

```diff
@@ -145,6 +145,26 @@ public void run() {
     }
 
     /**
+     * Starts the reporter polling at the given period.
```

Comment:
Also, the tests are not isolated. The collaborators (mocks) should be initialized per test method, not per class.